src/py-opentimelineio/opentimelineio/adapters/svg.py

```diff
@@ -0,0 +1,1017 @@
+                         34, 26, 36, 36, 28, 36, 36, 36,
+                         36, 36, 36, 36, 36, 27, 27, 36, 35, 36, 35, 65, 42, 43, 42,
+                         44, 35, 34, 43, 46, 25, 39, 40,
+                         31, 59, 47, 43, 41, 43, 44, 39, 28, 44, 43, 65, 37, 39, 34,
+                         37, 42, 37, 50, 37, 32, 43, 43,
+                         39, 43, 40, 30, 42, 45, 23, 25, 39, 23, 67, 45, 41, 43, 42,
+                         30, 40, 28, 45, 33, 52, 33, 36,
+                         31, 39, 26, 39, 55]
+image_width = 1000.0
+image_height = 660.0
+font_family = "Roboto"
+lines = ['']
+
+
+def convert_point_to_svg_coordinates(point=Point(0, 0)):
+    y = image_height - point.y
+    return Point(point.x, y)
+
+
+def convert_rect_to_svg_coordinates(rect=Rect()):
+    """Convert to SVG coordinate system (0,0 at top-left)"""
+    normalized_rect = rect.normalized()
+    normalized_rect.origin = convert_point_to_svg_coordinates(
+        normalized_rect.origin)
+    normalized_rect.height *= -1
+    return normalized_rect.normalized()
+
+
+def svg_color(color):
+    return 'rgb({},{},{})'.format(repr(color[0]), repr(color[1]), repr(color[2]))
+
+
+def draw_rect(rect, stroke_width=2.0, stroke_color=COLORS['black']):
+    svg_rect = convert_rect_to_svg_coordinates(rect)
+    rect_str = r'<rect x="{}" y="{}" width="{}" height="{}" ' \
+               r'style="fill:rgb(255,255,255);stroke-width:{};' \
+               r'stroke:{};opacity:1;' \
+               r'fill-opacity:0;" />'.format(repr(svg_rect.origin.x),
+                                             repr(svg_rect.origin.y),
+                                             repr(svg_rect.width),
+                                             repr(svg_rect.height),
+                                             repr(stroke_width),
+                                             svg_color(
+                                                 stroke_color))
+    lines.append(rect_str)
+
+
+def draw_dashed_rect(rect, stroke_width=2.0, stroke_color=COLORS['black'],
+                     fill_color=COLORS['white']):
+    svg_rect = convert_rect_to_svg_coordinates(rect)
+    rect_str = r'<rect x="{}" y="{}" width="{}" height="{}" ' \
+               r'stroke-dasharray="5" ' \
+               r'style="fill:{};stroke-width:{};' \
+               r'stroke:{};opacity:1;' \
+               r'fill-opacity:{};" />'.format(repr(svg_rect.origin.x),
+                                              repr(svg_rect.origin.y),
+                                              repr(svg_rect.width),
+                                              repr(svg_rect.height),
+                                              svg_color(fill_color),
+                                              repr(stroke_width),
+                                              svg_color(
+                                                  stroke_color),
+                                              repr(fill_color[3] / 255.0))
```

Comment:
Consider using ETree to build XML: https://docs.python.org/3/library/xml.etree.elementtree.html#building-xml-documents

This lets you build elements using code like `ET.Element("rect", {"x": str(svg_rect.origin.x), "y": str(svg_rect.origin.y), "another attribute": some_string_value})`. These can be composed into the SVG document hierarchy and serialized to a string using `ET.tostring`. This means that you don't have to worry about writing code for cases like when "<" is used.

In `draw_text` this could be especially helpful (note the auto-conversion to escaped character sequences):
```python
elem = ET.Element("text", {"x": "1", "y": "1"})
elem.text = "<something unexpectéd>"
ET.tostring(elem)
'<text x="1" y="1">&lt;something unexpect&#233;d&gt;</text>'
```